### DISCUSSION TITLE
"Start Ragdolling" in blueprint crashes Unreal

### DISCUSSION THREAD

**Discussion by woodencase01** [Sixze/ALS-Refactored#315] [Q&A]
I'm having a hard time using the ragdoll in my video game with ALS Refactored as the game crash.
I've simplified my test down to the basic ALS character, implementing the health component with a simple binding:
![image](https://github.com/Sixze/ALS-Refactored/assets/4787041/86ea73f1-7848-4de2-8480-4d235c79fb71)

As soon as the bullet hit and the health goes to zero, the character fire the event "HandleDeath" and start ragdolling, then the engine crash:

https://github.com/Sixze/ALS-Refactored/assets/4787041/7c8a8f40-694c-48ba-b3f3-8781d51c366a

With the following error code:
```
Assertion failed: BoneIndex < BoneSpaceTransforms.Num() [File:D:\build\++UE5\Sync\Engine\Source\Runtime\Engine\Private\PhysicsEngine\PhysAnim.cpp] [Line: 755]
UnrealEditor_Engine
UnrealEditor_ALS!AAlsCharacter::StartRagdollingImplementation() [E:\GitHub\TheBotanist\Plugins\ALS-Refactored\Source\ALS\Private\AlsCharacter_Actions.cpp:596]
UnrealEditor_CoreUObject
```

Using a trigger box in the level blueprint and calling StartRagdolling works:

https://github.com/Sixze/ALS-Refactored/assets/4787041/d2de12b9-e3e2-4d73-b94f-58e9edf4d5ff

Bypassing the "StartRagdolling" with a debug message works fine, so it's not the health logic:
![image](https://github.com/Sixze/ALS-Refactored/assets/4787041/d5af71e1-4eeb-48b9-b487-3db7913f420d)

Calling "StartRagdolling" from the level blueprint seems to work fine, but calling it from the "Handle Death" event taken from the health component doesn't work. I'm clueless about this crash! Your help is greatly appreciated.

**Comment by Sixze** [comment]
Do you have the latest plugin changes from the repository? This should have been fixed.

**Reply by woodencase01** [reply]
No, I'm still running on the 4.11
I will try it tonight, thanks for the feedback!

**Reply by woodencase01** [reply]
I've just tried compiling, and get the following errors:

```
E:\GitHub\TheBotanist\Plugins\ALS-Refactored\Source\ALSEditor\ALSEditor.Build.cs(8,51): error CS0117: 'EngineIncludeOrderVersion' does not contain a definition for 'Unreal5_2'
E:\GitHub\TheBotanist\Plugins\ALS-Refactored\Source\ALSExtras\ALSExtras.Build.cs(8,51): error CS0117: 'EngineIncludeOrderVersion' does not contain a definition for 'Unreal5_2'
E:\GitHub\TheBotanist\Plugins\ALS-Refactored\Source\ALSCamera\ALSCamera.Build.cs(8,51): error CS0117: 'EngineIncludeOrderVersion' does not contain a definition for 'Unreal5_2'
E:\GitHub\TheBotanist\Plugins\ALS-Refactored\Source\ALS\ALS.Build.cs(8,51): error CS0117: 'EngineIncludeOrderVersion' does not contain a definition for 'Unreal5_2'
Expecting to find a type to be declared in a target rules named 'TheBotanistEditorTarget'.  This type must derive from the 'TargetRules' type defined by Unreal Build Tool.
```
After replacing Unreal5_2 to Unreal5_1, I get the following error: 
`E:\GitHub\TheBotanist\Plugins\ALS-Refactored\Source\ALS\Public\Nodes\AlsRigUnits.h(7): Error: Unable to find parent struct type for 'FAlsRigVMFunction_ExponentialDecayVector' named 'FRigVMFunction_SimBase'`

I reverted the changes, and changed the line:
`GetMesh()->UpdateRBJointMotors();`
to

```
if (!GetMesh()->IsRunningParallelEvaluation() && GetMesh()->GetBoneSpaceTransforms().Num() > 0)
	{
		GetMesh()->UpdateRBJointMotors();
	}
```

And then this happened 🤩

https://github.com/Sixze/ALS-Refactored/assets/4787041/5852cef2-c368-41b8-be5c-e76ce950af1a

Thanks for the help @Sixze ! Is there any donation we can send you? I've been enjoying making video games and ALS-Refactored is big part of it, and your support too.
